src/main/java/org/betonquest/betonquest/compatibility/protocollib/hider/NPCHider.java

```diff
@@ -135,13 +140,29 @@ public void applyVisibility(final OnlineProfile onlineProfile, final Integer npc
         if (npc.isSpawned()) {
             final Set<ConditionID> conditions = npcs.get(npcID);
             if (conditions == null || conditions.isEmpty() || !BetonQuest.conditions(onlineProfile, conditions)) {
-                hider.showEntity(onlineProfile, npc.getEntity());
+                getEntityList(npc).forEach(entity -> hider.showEntity(onlineProfile, entity));
             } else {
-                hider.hideEntity(onlineProfile, npc.getEntity());
+                getEntityList(npc).forEach(entity -> hider.hideEntity(onlineProfile, entity));
             }
         }
     }
 
+    private List<Entity> getEntityList(final NPC npc) {
+        final List<Entity> entityList = new ArrayList<>();
+        entityList.add(npc.getEntity());
+
+        final HologramTrait hologramTrait = npc.getTraitNullable(HologramTrait.class);
```

Comment:
There is no other trait variable, so please use trait as variable name. Also, the name becomes false halfway through as it cannot be null anymore...